### PR TITLE
redmine task #704534 implementation

### PR DESCRIPTION
**Pull request by vlkff**
To deploy:
* pull in the code
• install modules one by one reiscout_map_feature -> reiscout_map -> reiscout_map_atlanta_content_feature , the content one as a last one after cache clearing
• probably you will need manually create Atlanta Map node
• probably you will need manually create Atlanta Map product with SKU 'all-atlanta-map-squares'
• enable the deploy module
• set price 10 p for square product on admin/config/reiscout-points/products-prices
• set price 1000 p for Atlanta product on admin/config/reiscout-points/products-prices
• enable points currency
• check is private filesystem set up
• cd to scripts and run the script from scripts dir to import squares from archive (pass squares dir path relative to docroot)
* uuid in reiscout_map_node_view() should be the one the Atlanta map node.

To test:
* create map square node with route files
* check that corresponding product created
* check that non-author and non admin has no access for file fields and files by their uri's 
* purchase the individual square
* check that you have access now and created order is fine
* check that after purchasing Atlanta map product you have access for all the squares now

### CODE REVIEW

File: docroot/sites/all/modules/content_features/reiscout_map_atlanta_content_feature/reiscout_map_atlanta_content_feature.features.inc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+/**
+ * Implements hook_node_info().
+ */
+function reiscout_map_atlanta_content_feature_node_info() {
+  $items = array(
+    'page' => array(
+      'name' => t('Basic page'),
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/content_features/reiscout_map_atlanta_content_feature/reiscout_map_atlanta_content_feature.features.inc]: nope, I will remove it

File: docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_base.inc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    'settings' => array(
+      'backend' => 'default',
+      'srid' => 4326,
+    ),
+    'translatable' => 0,
+    'type' => 'geofield',
+  );
+
+  // Exported field_base: 'field_map_sq_gpx'.
+  $field_bases['field_map_sq_gpx'] = array(
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_base.inc]: for that we need to create an s3 file system and mount it somewhere to public:// or private:// drupal filesystem? or how that gonna work ?

File: docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_instance.inc [hunk 3]
```diff
@@ -0,0 +1,700 @@
...
+    'entity_type' => 'node',
+    'field_name' => 'field_map_sq_preview',
+    'label' => 'Preview',
+    'required' => 0,
+    'settings' => array(
+      'alt_field' => 0,
+      'default_image' => 0,
+      'file_directory' => 'map-square/preview',
+      'file_extensions' => 'png gif jpg jpeg',
+      'max_filesize' => '2mb',
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_instance.inc]: a lot of images have size more then 1.4 mb.

File: docroot/sites/all/modules/reiscout/reiscout_map/SellMapSquareHelper.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+include_once drupal_get_path('module', 'reiscout_points_product') . '/SellEntityProductHelper/SellEntityProductHelper.php';
+
+include_once drupal_get_path('module', 'reiscout_points_product') . '/SellEntityProductHelper/SellEntityProductHelperInterface.php';
+
+include_once 'SellAtlantaMapSquaresHelper.php';
+
+class SellMapSquareHelper extends SellEntityProductHelper implements SellEntityProductHelperInterface {
+
+  protected $product_type = 'individual_map_square';
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/reiscout/reiscout_map/SellMapSquareHelper.php]: I got it =)

File: docroot/sites/all/modules/content_features/reiscout_map_atlanta_content_feature/reiscout_map_atlanta_content_feature.features.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/content_features/reiscout_map_atlanta_content_feature/reiscout_map_atlanta_content_feature.features.inc]: DONE

File: docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_instance.inc (hunk 3 above)
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_instance.inc]: I think the map preview image files required to be in high resolution so user could see the route with details before he purchase it

File: docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.inc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Implements hook_commerce_product_default_types().
+ */
+function reiscout_map_feature_commerce_product_default_types() {
+  $items = array(
+    'city_map_square' => array(
+      'type' => 'city_map_square',
+      'name' => 'City Square',
+      'description' => 'Buying this product user get an access to route data files attached for all map square nodes belongs to the city.',
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.inc]: DONE

File: docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.inc [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      'help' => '',
+      'revision' => 1,
+    ),
+    'individual_map_square' => array(
+      'type' => 'individual_map_square',
+      'name' => 'Individual Square',
+      'description' => 'Buying this product user get and access to route data files attached Individual map square node.',
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.inc]: DONE

File: docroot/sites/all/modules/reiscout/reiscout_map/SellMapSquareHelper.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/reiscout/reiscout_map/SellMapSquareHelper.php]: DONE

File: docroot/sites/all/modules/reiscout/reiscout_map/SellMapSquareHelper.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+include_once drupal_get_path('module', 'reiscout_points_product') . '/SellEntityProductHelper/SellEntityProductHelper.php';
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/reiscout/reiscout_map/SellMapSquareHelper.php]: DONE

File: docroot/sites/all/modules/reiscout/reiscout_map/SellAtlantaMapSquaresHelper.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  const SKU = 'all-atlanta-map-squares';
+
+  public function __construct() {
+    $product = commerce_product_load_by_sku(self::SKU);
+    if (empty($product)) {
+      throw new Exception(self::SKU . ' product is missed.');
+    }
+  }
+
+  public function get_product_type() {
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/reiscout/reiscout_map/SellAtlantaMapSquaresHelper.php]: DONE

File: docroot/sites/all/modules/reiscout/reiscout_map/SellAtlantaMapSquaresHelper.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      $account = $GLOBALS['user'];
+    }
+
+    if (reiscout_points_product_is_purchased(self::SKU, $account)) {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  public function get_product_form () {
```
[the remark this answers is not in the document]
Comment (vlkff) [reply docroot/sites/all/modules/reiscout/reiscout_map/SellAtlantaMapSquaresHelper.php]: DONE

File: docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_instance.inc (hunk 3 above)
[the remark this answers is not in the document]
Comment (dzutaro) [reply docroot/sites/all/modules/features/reiscout_map_feature/reiscout_map_feature.features.field_instance.inc]: Images can be compressed without noticeable reduction in image quality.

For example, this is the image that is used for Atlanta map (**914.74 KB**):
https://reiscout.com/sites/default/files/Atlanta%20Metro%20Image%20Map.png

And here is a compressed version of the image (**398.67 KB**):
https://reiscout.com/sites/default/files/Atlanta-Metro-Image-Map.jpg

**Pull request merged**